keras_eval/data_generators.py

```diff
@@ -0,0 +1,328 @@
+        if 'crop_original' in data_augmentation.keys():
+            self.crop_original = self.data_augmentation['crop_original']
+
+        self.transforms = ['none']
+        if 'transforms' in self.data_augmentation.keys():
+            if isinstance(self.data_augmentation['transforms'], list):
+                self.transforms += self.data_augmentation['transforms']
+            else:
+                raise ValueError('Incorrect format for `transforms`, a list of transforms is expected')
+        self.n_transforms = len(self.transforms)
+
+        if 'scale_sizes' in self.data_augmentation.keys():
+            if self.data_augmentation['scale_sizes'] == 'default':
+                self.scale_sizes = self._get_default_sizes(self.target_size[0])
+            elif isinstance(self.data_augmentation['scale_sizes'], list) and \
+                    all(isinstance(x, int) for x in self.data_augmentation['scale_sizes']):
+                self.scale_sizes = []
+                for size in self.data_augmentation['scale_sizes']:
+                    size = round(size)
+                    if size % 2 != 0:
+                        size += 1
+                    self.scale_sizes.append(size)
+            else:
+                raise ValueError('Incorrect format for `scale_sizes`, list of ints or `= default` is expected')
+
+            self.n_crops = len(self.scale_sizes) * 5 * 3 * self.n_transforms + self.n_transforms
+        else:
+            self.scale_sizes = None
+            self.n_crops = self.n_transforms
+
+    @staticmethod
+    def _check_data_augmentation_keys(data_augmentation):
+        if isinstance(data_augmentation, dict):
+            keys = data_augmentation.keys()
+            if 'scale_sizes' not in keys and 'transforms' not in keys and 'crop_original' not in keys:
+                raise ValueError('data_augmentation dictionary should contain '
+                                 '`crop_original` `scale_sizes` or `transforms` as keys')
+        else:
+            raise ValueError('`data_augmentation` is %s and it should be a dictionary' % type(data_augmentation))
+
+    @staticmethod
+    def _get_default_sizes(target_size, multipliers=(1.1, 1.2, 1.3, 1.4)):
+        sizes = []
+        for multiplier in multipliers:
+            size = round(target_size * multiplier)
+            if size % 2 != 0:
+                size += 1
+            sizes.append(size)
+        return sizes
+
+    @staticmethod
+    def _get_3_crops(image):
+        w, h = image.size
+        w_center = w / 2
+        h_center = h / 2
+
+        if w >= h:
+            im_1 = image.crop((0, 0, h, h))
+            im_2 = image.crop((w_center - h / 2, 0, w_center + h / 2, h))
+            im_3 = image.crop((w - h, 0, w, h))
+        else:
+            im_1 = image.crop((0, 0, w, w))
+            im_2 = image.crop((0, h_center - w / 2, w, h_center + w / 2))
+            im_3 = image.crop((0, h - w, w, h))
+
+        image.save('ori.jpg')
```

Comment:
Why do you save the original image? To pull them all from the same folder?

Reply:
no, it was for debugging, forgot to remove it